fluentnamer/src/main/java/com/azure/autorest/fluent/transformer/SchemaCleanup.java

```diff
@@ -76,13 +76,16 @@ private static boolean tryCleanup(CodeModel codeModel, Set<String> javaNamesForP
                         String name = Utils.getJavaName(o);
                         return FluentType.nonSystemData(name) && FluentType.nonManagementError(name);
                     })
-                    .flatMap(s -> s.getProperties().stream()
+                    .flatMap(s -> Stream.concat(
+                            Stream.of(s),
+                            s.getProperties()
+                                    .stream()
 //                                    .filter(Utils::nonFlattenedProperty)
-                                    .map(Property::getSchema)
-                                    .map(SchemaCleanup::schemaOrElementInCollection)
-                                    .filter(Objects::nonNull)
-                                    .filter(s1 -> !Objects.equals(s, s1))   // schema of property is not the same of itself, solve the simplest recursive reference case
-                    )
+                                    .map(Property::getSchema)))
+                    .flatMap(s -> itselfAndAllParents(s)
+                            .map(SchemaCleanup::schemaOrElementInCollection)
```

Comment:
The schemaInUse should be the elementType of the array/dict. I didn't see code about that.

Reply:
Yeah, it's better to get all parents(Dict, Array) of current schemaInUse. Changed.